module_auto_update/models/module.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import logging
+
+from odoo import api, fields, models
+from odoo.modules.module import get_module_path
+
+_logger = logging.getLogger(__name__)
+try:
+    from checksumdir import dirhash
+except ImportError:
+    _logger.debug('Cannot `import checksumdir`.')
+
+
+class Module(models.Model):
+    _inherit = 'ir.module.module'
+
+    checksum_installed = fields.Char(
+        string="Installed Module Checksum",
+    )
+    checksum_dir = fields.Char(
+        string="Current Module Checksum",
+        compute='_compute_checksum_dir',
+    )
+    upgrade_available = fields.Boolean(
+        default=False,
+    )
+
+    @api.multi
+    def _compute_checksum_dir(self):
+        for r in self:
+            r.checksum_dir = dirhash(get_module_path(r.name), 'sha1')
```

Comment:
I think you're correct here Jairo. The `pyc` files contain a `moddate` header, which is the date of compilation as I understand it. Due to this, the hash can and will change when the file is recomputed, regardless of whether there were actually any changes.

Luckily `checksumdir` has an `excluded_extension` key, which interestingly enough has `pyc` in it as the example:

```
sha256hash = dirhash(directory, 'sha256', excluded_extensions=['pyc'])
```

IMO we should use an array declared at class level to determine the extensions that should be excluded. We could also use a method, but I think that's a bit overengineered